/*
 * Copyright (c) 2006, 2007, Oracle and/or its affiliates. All rights reserved.
 *
 * Permission is hereby granted, free of charge, to any person obtaining a
 * copy of this software and associated documentation files (the "Software"),
 * to deal in the Software without restriction, including without limitation
 * the rights to use, copy, modify, merge, publish, distribute, sublicense,
 * and/or sell copies of the Software, and to permit persons to whom the
 * Software is furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice (including the next
 * paragraph) shall be included in all copies or substantial portions of the
 * Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL
 * THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
 * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
 * DEALINGS IN THE SOFTWARE.
 *
 */

Solaris 9 shipped with IPv6 support based on the X.Org draft standard, which 
used "inetv6" - the final standard renamed it to "inet6", but since we already
shipped, we retain the old name for backwards compatibility.  
(See PSARC 2004/285.)

diff -urp -x '*~' -x '*.orig' Xtranssock.c Xtranssock.c
--- Xtranssock.c	2010-10-29 18:21:40.000000000 -0700
+++ Xtranssock.c	2011-02-11 16:46:56.023653158 -0800
@@ -188,6 +188,9 @@ static Sockettrans2dev Sockettrans2devta
     {"tcp",AF_INET6,SOCK_STREAM,SOCK_DGRAM,0},
     {"tcp",AF_INET,SOCK_STREAM,SOCK_DGRAM,0}, /* fallback */
     {"inet6",AF_INET6,SOCK_STREAM,SOCK_DGRAM,0},
+#ifdef SUNSOFT /* Backwards compatibility with Solaris 9 */
+    {"inetv6",AF_INET6,SOCK_STREAM,SOCK_DGRAM,0},
+#endif   
 #endif
 #endif /* TCPCONN */
 #ifdef UNIXCONN
